examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/CastingServerBridge.mm

```diff
@@ -118,10 +109,157 @@ - (instancetype)init
     return self;
 }
 
+/**
+ Dispatches the block on the Matter SDK dispatch queue.
+
+ @param description Optional. A description of the callback to be dispatched for diagnostic purposes.
+ @param block The Block to be dispatched.
+ */
+- (void) dispatchOnMatterSDKQueue:(const NSString * _Nullable)description
+                            block:(dispatch_block_t)block
+{
+    if (nil != description) {
+        ChipLogProgress(AppServer, "[SYNC] CastingServerBridge %s", [description UTF8String]);
+    }
+    dispatch_async(_chipWorkQueue, ^{
+        // The Matter SDK is not, generally, internally thread-safe. It relies upon the client to lock
+        // the SDK prior to any interactions after chip::DeviceLayer::PlatformMgr().InitChipStack()
+        //
+        // Note that it is presently safe to do at this point because InitChipStack is called in the
+        // constructor for the CastingServerBridge.
+        chip::DeviceLayer::StackLock lock;
+
+        if (nil != description) {
+            ChipLogProgress(AppServer, "[ASYNC (CHIP)] CastingServerBridge BEGIN %s", [description UTF8String]);
+        }
+        block();
+        if (nil != description) {
+            ChipLogProgress(AppServer, "[ASYNC (CHIP)] CastingServerBridge END %s", [description UTF8String]);
+        }
+    });
+}
+
+/**
+ Dispatches a client callback via a `dispatch_block_t` to be run on the client's `dispatch_queue_t`.
+
+ @param description Optional. A description of the callback to be dispatched for diagnostic purposes.
+ @param queue The client's Dispatch Queue.
+ @param block The Block to be invoked to invoke the client callback.
+ */
+- (void) dispatchOnClientQueue:(const NSString * _Nullable )description
+                          queue:(dispatch_queue_t)queue
```

Comment:
super nit, might be nice for this API to have queue either as the first arg or the last arg, its kind of weird to have it in the middle since you have the `dispatchOnMatterSDKQueue` function listed with `description, block`

not a huge deal though

Reply:
Good point, I'll move the queue to the first argument. The last argument is reserved for the block based on [the Objective C guidelines](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithBlocks/WorkingwithBlocks.html#//apple_ref/doc/uid/TP40011210-CH8-SW7).